myst_parser/main.py

```diff
@@ -186,8 +201,9 @@ def default_parser(config: MdParserConfig) -> MarkdownIt:
         md.use(deflist_plugin)
     if "substitution" in config.enable_extensions:
         md.use(substitution_plugin, *config.sub_delimiters)
-    if config.heading_anchors is not None:
-        md.use(anchors_plugin, max_level=config.heading_anchors)
+    if use_anchors_plugin:
+        max_level = config.heading_anchors if config.heading_anchors is not None else 2
```

Comment:
I don't think this is necessary, and will lead to ambiguity. Only `heading_anchors` should be used to set the level.

Reply:
2 is the [default](https://github.com/executablebooks/mdit-py-plugins/blob/master/mdit_py_plugins/anchors/index.py#L12), but I can build a kwargs dict instead and not pass max_level if this config isn’t set

Reply:
I just don't feel that adding a function should affect whether it is enabled or disabled

Reply:
I've made a commit to change the behavior.

Reply:
Cheers